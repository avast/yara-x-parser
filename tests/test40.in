rule test {
    condition:
        foo.bar[0].baz == foo.bar.baz()
}