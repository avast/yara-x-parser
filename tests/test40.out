SOURCE_FILE@0..68
  RULE@0..68
    RULE_KW@0..4 "rule"
    WHITESPACE@4..5 " "
    IDENTIFIER@5..9 "test"
    WHITESPACE@9..10 " "
    BLOCK_EXPR@10..68
      L_BRACE@10..11 "{"
      WHITESPACE@11..16 "\n    "
      CONDITION@16..66
        CONDITION_KW@16..25 "condition"
        COLON@25..26 ":"
        WHITESPACE@26..35 "\n        "
        BOOLEAN_TERM@35..66
          EXPRESSION@35..66
            EXPRESSION@35..49
              INDEXING_EXPR@35..45
                PRIMARY_EXPR@35..42
                  IDENTIFIER@35..38 "foo"
                  DOT@38..39 "."
                  IDENTIFIER@39..42 "bar"
                L_BRACKET@42..43 "["
                PRIMARY_EXPR@43..44
                  INT_LIT@43..44 "0"
                R_BRACKET@44..45 "]"
              DOT@45..46 "."
              PRIMARY_EXPR@46..49
                IDENTIFIER@46..49 "baz"
            WHITESPACE@49..50 " "
            EQ@50..52 "=="
            WHITESPACE@52..53 " "
            FUNCTION_CALL_EXPR@53..66
              PRIMARY_EXPR@53..64
                IDENTIFIER@53..56 "foo"
                DOT@56..57 "."
                IDENTIFIER@57..60 "bar"
                DOT@60..61 "."
                IDENTIFIER@61..64 "baz"
              L_PAREN@64..65 "("
              R_PAREN@65..66 ")"
      WHITESPACE@66..67 "\n"
      R_BRACE@67..68 "}"
